samples/dnn/yolo_detector.cpp

```diff
@@ -131,25 +133,28 @@ void yoloPostProcessing(
         // remove the second element
         outs.pop_back();
         // unsqueeze the first dimension
-        outs[0] = outs[0].reshape(0, std::vector<int>{1, 8400, 84});
+        outs[0] = outs[0].reshape(0, std::vector<int>{1, 8400, nc + 4});
```

Comment:
8400 is also depends on number of classes? So should it be `(nc + 4) * 100` or not?

Reply:
`nc` seems redundant as it can be computed from original shape:
```cpp
int nc = outs[0].total() / 8400 - 4;
```
or, if 8400 is also depends on `nc`:
```cpp
int nc = sqrt(outs[0].total() / 100) - 4;
```

Reply:
> 8400 is also depends on number of classes? So should it be `(nc + 4) * 100` or not?

8400 depends on number of anchors only and has nothing to do with `nc`. 
`nc` depends on which dataset detector what trained. On `COCO` dataset `nc = 80`. For other datasets it might be different.

Reply:
Got it, thanks! So do you mind to compute `nc` from total number of `outs[0]` elements?

Reply:
I strongly believe that it should stay as papermeter. As this is a parameter during traning. Computing it via `outs[0]` is kind of hack.

Reply:
The corner case is a background class. Some training utils may include this as a separate class - some not. So even if you trained model for 80 real classes, there is a variation nc=80 nc=81 and it may confuse even more.

Reply:
Computing `nc` using number of anchor points is not good since number of anchors depens on the image size. Inference can happen on different image resolutons, depending on use case.